torch/csrc/Device.cpp

```diff
@@ -137,6 +137,24 @@ PyObject *THPDevice_rc(PyObject *a, PyObject *b, int op) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject *THPDevice_reduce(THPDevice *self)
+{
+  auto ret = THPObjectPtr{PyTuple_New(2)};
+  if (!ret) throw python_error();
+
+  py::object torch_module = py::module::import("torch");
+  py::object torch_device = torch_module.attr("device");
+  PyTuple_SET_ITEM(ret.get(), 0, torch_device.release().ptr());
+
+  if (self->device.is_default) {
+    PyTuple_SET_ITEM(ret.get(), 1, Py_BuildValue("(s)", deviceTypeString(self->device.type)));
```

Comment:
You need to check for the `NULL` return from `Py_BuildValue`. I think ignoring it and returning a half-constructed tuple might lead to segfaults.